plugins/woocommerce/tests/api-core-tests/tests/products/products-crud.test.js

```diff
@@ -1098,25 +1098,183 @@ test.describe('Products API tests: CRUD', () => {
 		});
 	});
 
-	test('can add a variable product', async ({
-		request
-	}) => {
-		const response = await request.post('wp-json/wc/v3/products', {
-			data: variableProduct,
+	test.describe('Product variation tests: CRUD', () => {
+		let variableProductId;
+		let productVariationId;
+
+		test('can add a variable product', async ({
+			request
+		}) => {
+			const response = await request.post('wp-json/wc/v3/products', {
+				data: variableProduct,
+			});
+			const responseJSON = await response.json();
+			variableProductId = responseJSON.id;
+			expect(response.status()).toEqual(201);
+			expect(typeof variableProductId).toEqual('number');
+			expect(responseJSON).toMatchObject(variableProduct);
+			expect(responseJSON.status).toEqual('publish');
 		});
-		const responseJSON = await response.json();
-		const variableProductId = responseJSON.id;
 
-		expect(response.status()).toEqual(201);
-		expect(typeof variableProductId).toEqual('number');
-		expect(responseJSON).toMatchObject(variableProduct);
-		expect(responseJSON.status).toEqual('publish');
+		test('can add a product variation', async ({
+			request
+		}) => {
+			const response = await request.post(`wp-json/wc/v3/products/${variableProductId}/variations`, {
+				data: {
+					"regular_price": "29.00",
+					"attributes": [{
+						"name": "Colour",
+						"option": "Green"
+					}]
+				},
+			});
+			const responseJSON = await response.json();
+			productVariationId = responseJSON.id;
+			expect(response.status()).toEqual(201);
+			expect(typeof productVariationId).toEqual('number');
+			expect(responseJSON.regular_price).toEqual("29.00");
+		});
 
-		// Cleanup: Delete the variable product
-		await request.delete(`wp-json/wc/v3/products/${ variableProductId }`, {
-			data: {
-				force: true,
-			},
+		test('can retrieve a product variation', async ({
+			request
+		}) => {
+			const response = await request.get(`wp-json/wc/v3/products/${variableProductId}/variations/${productVariationId}`);
+			const responseJSON = await response.json();
+			expect(response.status()).toEqual(200);
+			expect(responseJSON.id).toEqual(productVariationId);
+			expect(responseJSON.regular_price).toEqual('29.00');
+		});
+
+		test('can retrieve all product variations', async ({
+			request
+		}) => {
+			// call API to retrieve all product tags
```

Comment:
These comments refer to product tags, but we're working with variations here--so we need to update these accordingly. This line and `1162`, `1177`, and `1187` looks like all need to be updated.

Reply:
Fully agree @zhongruige! This file is getting large. A consideration is do we break it out into separate files and do some extra data setup in each of these files.. I'll mull it over and ping you on this.

Also - Thanks for pointing out the comments - I actually noticed a couple of others that I will update also - the shame of a copy-paste 😉

Reply:
Sounds good @nigeljamesstevenson! Thanks for getting in these updates.